### PR TITLE
Add a system name auto-correction method

### PR DESCRIPTION
**Pull request by Eearslya**
This PR adds in a new method to Ratlib, `correct_system_name`. This method performs a similar task as Mecha2's autocorrection, where it will take a procedurally generated system name and ensure that it follows the proper format, substituting letters with numbers and vice-verse where appropriate.

### CODE REVIEW

File: utils/ratlib.py [hunk 1]
```diff
@@ -188,6 +188,39 @@ def duration(time: datetime.timedelta) -> str:
     return humanfriendly.format_timespan(time, detailed=False, max_units=2)
 
 
+def correct_system_name(system: str) -> str:
```
Comment (MHajoha) [inline utils/ratlib.py]: Can we stop piling unrelated functions in `ratlib.py`? If @theunkn0wn1 agrees, I'd like you to move this into its own module under `utils/`.

Comment (theunkn0wn1) [reply utils/ratlib.py]: I would argue `ratlib.py` is horribly named if this function is unrelated.

According to the module docstring its for
```
ratlib.py - IRC/Fuelrat nickname utilities
```
yet its named as `ratlib`. 

Therefore, the docstring and module's name don't match, which leads to the confusion on what the file is actually for.
`ratlib` suggests is a generic library`, the docstring suggests its specifically for nickname utilities.

Comment (MHajoha) [reply utils/ratlib.py]: Regardless of the docstring, I see the problem in collecting unrelated functions and classes in a single module. It's not a crime to have modules containing only a single function or class, you know.

Comment (Eearslya) [reply utils/ratlib.py]: Resolved via its own file.

**Pull request merged**